wcfsetup/install/files/lib/system/session/SessionHandler.class.php

```diff
@@ -339,9 +312,54 @@ protected function defineConstants() {
 	 * Initializes security token.
 	 */
 	protected function initSecurityToken() {
-		if ($this->getVar('__SECURITY_TOKEN') === null) {
-			$this->register('__SECURITY_TOKEN', bin2hex(\random_bytes(20)));
+		$xsrfToken = '';
+		if (!empty($_COOKIE['XSRF-TOKEN'])) {
+			// We intentionally do not extract the signed value and instead just verify the correctness.
+			//
+			// The reason is that common JavaScript frameworks can use the contents of the `XSRF-TOKEN` cookie as-is,
+			// without performing any processing on it, improving interoperability. Leveraging that JavaScript framework
+			// feature requires the author of the controller to check the value within the `X-XSRF-TOKEN` request header
```

Comment:
"Leveraging that JavaScript framework feature requires" There is at least an article missing.

Reply:
It reads correctly to me. Is the following phrasing better: “To leverage that JavaScript framework feature the author needs to …”?

Reply:
Oh, I now understand the sentence. It should read "Leveraging *this* JavaScript framework feature requires", then.